js/src/product-feed/issues-table-card/getActiveIssueType.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { getQuery } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { ISSUE_TYPE_ACCOUNT } from '.~/constants';
+
+/**
+ * Gets the active Issue Type Filter based on the `issueType` query property.
+ *
+ * @return {string} The active Issue Type Filter. If `issueType` is not available returns Account Issue Type by default.
+ */
+const getActiveIssueType = () => getQuery()?.issueType ?? ISSUE_TYPE_ACCOUNT;
```

Comment:
❓ same as https://github.com/woocommerce/google-listings-and-ads/pull/1176#discussion_r777949997

Reply:
Changed here --> https://github.com/woocommerce/google-listings-and-ads/pull/1176/commits/18028089dd0a65715fddac9aa840cfcb5cd32a2b

Reply:
Btw notice that is not possible to get issueType as 0 (except if you're messing around with the params in the URL, and even tho, not sure because I think placing a 0 will be received as "0", but not sure). 

On the other side, It is possible to have it `undefined`. (see Navigation) Hence it would be more descriptive to have null check. This change could confuse developers in a way of "thinking it could have 0 or NaN values)  

However I'm fine with || since is not painful neither.

Reply:
The `||` operator divides the values into truthy and falsy groups, and it's commonly used to set the default value if the valid/invalid could be easily grouped by truthy and falsy. The context of `??` operator more likely mean that only `null`, and `undefined` are invalid values and the rest values are valid.

So, I'd suggest using `??` operator when the values are exactly treating all or at least one of  `''`, `0`, `NaN`, `false` as valid. Otherwise, the `||` operator should be more proper.

Reply:
- AFAIK `NaN` case is never going to happen since getQuery returns a string, hence, it is always NaN
-  hence bool false and int 0, they are never going to happen, since getQuery parse the values as strings.  
- Strictly talking about '0' and 'false' (which are falsy in JS...) they are valid values if somehow the issue type filter is named as 'false' or '0', hence, strictly talking they are valid values, as it is 'test' or any other string outside 'account or 'product'.  

In the current context the only acceptable values are undefined, 'product' and 'account'. And that was my attempt to describe in the code (obviously I failed on that)

Being super strict about types. It should be `return getQuery()?.issueType === ISSUE_TYPE_PRODUCT ? ISSUE_TYPE_PRODUCT : ISSUE_TYPE_ACCOUNT;`  

or with scalability in mind...

```
const AVAILABLE_ISSUES = [...] // constant list with available issues types
const issueType = getQuery()?.issueType;
return  AVAILABLE_ISSUES.includes(issueType) ? issueType : ISSUE_TYPE_ACCOUNT;
```  

Since these 2 new suggestions feels like a bit of over-engineering for me, I'm fine with `||` as is suggested and committed after your comment.

Reply:
> * AFAIK `NaN` case is never going to happen since getQuery returns a string, hence, it is always NaN
> * hence bool false and int 0, they are never going to happen, since getQuery parse the values as strings.

> In the current context the only acceptable values are undefined, 'product' and 'account'. And that was my attempt to describe in the code (obviously I failed on that)

I guess the difference in perspective is that I was concerned with what's the valid value of issue type, rather than the value range returned by `getQuery()`.

Most precisely, when the valid values are only `'account'` and `'product'`. Then no matter what `getQuery()` or any other codes that produce a value for checking, the result of `||` will be more correct than `??`. Unlike `??`, which leads to further thought about whether the value to be evaluated might be falsy other than `null` and `undefined`, resulting in an unintended issue type.

Surely the `||` is not completely correct, but it's at least more accurate than `??` and does not lead to additional checks as described above if a developer is not familiar with `getQuery()`.

> * Strictly talking about '0' and 'false' (which are falsy in JS...) they are valid values if somehow the issue type filter is named as 'false' or '0', hence, strictly talking they are valid values, as it is 'test' or any other string outside 'account or 'product'.

I'm not quite sure whether you meant the strings `'0'` and `'false'` as falsy in JavaScript? If yes, I have to point out they are truthy values in JavaScript in fact. And I don't understand the association of ''test'' with the context.

Regardless, even if `'0'` and `'false'` were valid values, it could still get the correct results with `||`.

> Since these 2 new suggestions feels like a bit of over-engineering for me, I'm fine with `||` as is suggested and committed after your comment.

Actually, I haven't suggested using `||`. The comparing discussion between `||` and `??` for this case was followed your comment. I would choose a solution like `AVAILABLE_ISSUES.includes(issueType)`. I don't think it is an over-engineering approach since it is the most robust implementation and brings the most accurate results.

In conclusion, I'm ok with `||` because it's very common to adopt in such cases. Just like, if falsy value `''`, `0`, or etc was also a valid value, I'd think it's ok to adopt `??`.

Reply:
Thanks

> I'm not quite sure whether you meant the strings '0' and 'false' as falsy in JavaScript? If yes, I have to point out they are truthy values in JavaScript in fact. And I don't understand the association of 'test' with the context.

My bad, I messed with my explanation... what I meant is that since '0' or 'false' cases are not treated as falsy they are evaluated as if you pass 'test' or any other string beside 'product' and 'account' as the `issueType` query value 


Resolution: 

Since now the default value is going to change to a function depending on the totals. 
(see pcTzPl-da-p2#comment-430) 

We will probably promote this to a hook or at least modify it, at that point I will implement `AVAILABLE_ISSUES.includes(issueType)` as suggested